code/modules/admin/artifacts_panel.dm

```diff
@@ -0,0 +1,109 @@
+/datum/admins/proc/artifacts_panel()
+	if (!(SSxenoarch?.initialized))
+		to_chat(usr,"<span class='danger'>The Xenoarch subsystem hasn't initialized yet!</span>")
+
+	else if (!SSxenoarch?.artifact_spawning_turfs.len)
+		to_chat(usr,"<span class='danger'>The Xenoarch subsystem seems to not have spawned any large artifact in the map. Lack of valid asteroid turfs?</span>")
+
+	var/dat = {"<html>
+		<head>
+		<style>
+		table,h2 {
+		font-family: Arial, Helvetica, sans-serif;
+		border-collapse: collapse;
+		}
+		td, th {
+		border: 1px solid #dddddd;
+		padding: 8px;
+		}
+		tr:nth-child(even) {
+		background-color: #dddddd;
+		}
+		</style>
+		</head>
+		<body>
+		<h2 style="text-align:center">Artifact Panel</h2>
+		<table>
+		<tr>
+		<th style="width:1%">Artifact ID</th>
+		<th style="width:1%">Status</th>
+		<th style="width:3%">Type</th>
+		<th style="width:1%">Primary Effect</th>
+		<th style="width:1%">Secondary Effect</th>
+		</tr>
+		"}
+
+	//First we list every large artifact that exists currently in the world.
+	//non-anomaly artifacts such as Mysterious Cloning Pod and Replicator that were admin-spawned won't appear here as they do not have an Artifact ID, unless they get analyzed later.
+	var/list/artifacts_checked_already = list()
+
+	for(var/ID in excavated_large_artifacts)
+		artifacts_checked_already += ID
+		var/atom/A = excavated_large_artifacts[ID]
+		if (!(A?.loc))
+			if (!(ID in destroyed_large_artifacts))
+				var/list/artifact_data = list(
```

Comment:
This being a list makes the code painful to read, I would suggest turning it into a datum. At the very least, if you intend on keeping it a list, the indices should be `#define`d and used appropriately, but that's just less readable than a datum for no evident benefit.

Reply:
I was probably tired and went for the first solution that my mind came up with but a datum is indeed quite a better choice. I've rewritten it so that it now also works for non random artifacts. Just gotta add a single line to their Destroy().

Reply:
0% tested but it should work

Reply:
It's still a list, though? Maybe test it more than 0%

Reply:
oh I forgot that part

Reply:
Should be good now, still my friend arrives in 3 minutes so I have no more time to test today.